data/json/statistics.json

```diff
@@ -1756,5 +1756,12 @@
     "stat_type": "count",
     "event_transformation": "avatar_reads_book",
     "description": { "str_sp": "read a book" }
+  },
+  {
+    "id": "num_bacon",
+    "type": "event_statistic",
+    "stat_type": "count",
+    "event_type": "ate_bacon",
+    "description": { "str": "bacon consumed", "str_pl": "bacons consumed" }
```

Comment:
You can avoid the cpp changes by using the already available event transformation avatar_consumes_item
Something like
```json
{
  "id": "avatar_consumes_bacon",
  "type": "event_transformation",
  "event_transformation": "avatar_consumes_item",
  "value_constraints": { "itype": { "equals": [ "itype_id", "bacon" ] } },
  "drop_fields": [ "itype" ]
},
{
  "id": "num_avatar_consumes_bacon",
  "type": "event_statistic",
  "stat_type": "count",
  "event_transformation": "avatar_consumes_bacon",
  "description": { "str": "piece of bacon consumed", "str_pl": "pieces of bacon consumed" }
},
```
instead should simplify things and let this be a json only pr